task/manager.go

```diff
@@ -105,11 +107,14 @@ type Manager struct {
 	cluster Cluster
 	// queue of actions.
 	queue chan func()
+	// collector registry.
+	collector map[string]*LogCollector
```

Comment:
**hint**: Keep a map of collectors to ensure only one is started for each container.